src/reaper_osara.cpp

```diff
@@ -4025,6 +4025,61 @@ void cmdMoveItemsOrEnvPoint(Command* command) {
 	}
 }
 
+void cmdInsertOrMoveSpecificMarker(Command* command) {
+	int cmd = command->gaccel.accel.cmd;
+	int beforeCount = CountProjectMarkers(0, nullptr, nullptr);
+	int wantNum;
+	// Work out the desired marker based on the command ID.
+	if (cmd == 40656)
+		wantNum = 10;
+	else if (40657 <= cmd && cmd <= 40665)
+		wantNum = cmd - 40656;
+	else
+		return; // Shouldn't happen.
+	double beforePos = -1;
+	bool markerExistsBefore = false;
+	for (int i = 0; i < beforeCount; ++i) {
+		double pos;
+		bool reg;
+		int num;
+		EnumProjectMarkers(i, &reg, &pos, nullptr, nullptr, &num);
+		if (reg || wantNum != num)
+			continue;
+		beforePos = pos;
+		markerExistsBefore = true;
+		break;
+	}
+	Main_OnCommand(cmd, 0);
+	int afterCount = CountProjectMarkers(0, nullptr, nullptr);
+	for (int i = 0; i < afterCount; ++i) {
+		bool reg;
+		int num;
+		const char* name;
+		EnumProjectMarkers(i, &reg, nullptr, nullptr, &name, &num);
+		if (num != wantNum || reg)
+			continue;
+		fakeFocus =FOCUS_MARKER;
+		ostringstream s;
+		if (name[0]) {
+			// Translators: used when reporting a named marker has been moved. {} will be
+			// replaced with the name of the marker; e.g. "v2 marker moved"
+			s << format(translate("{} marker moved"), name);
+		} else{
+			if (!markerExistsBefore) {
+				// Translators: Reports an unnamed marker has been inserted. {} is replaced with the marker number.
+				s << format(translate("marker {} inserted"), num);
+			} else if (beforePos == -1) {
```

Comment:
1. Can beforePos be -1 if markerExistsBefore is true? In what situation?
2. If not, can we get rid of markerExistsBefore and just use beforePos -1 to detect this case?
3. If beforePos can be -1 in some other situation, can we combine these two checks into one to avoid code duplication? That is, `if (!markerExistsBefore || beforePos == -1)`